src/DynamoCore/Search/SearchDictionary.cs

```diff
@@ -323,11 +323,17 @@ private static string[] SplitOnWhiteSpace(string s)
             return s.Split(null);
         }
 
-        private static bool ContainsSpecialCharacters(string element)
+        public static Char[] SpecialAndInvalidCharacters()
         {
-            return element.Contains("*") || element.Contains(".") || element.Contains(" ")
-                || element.Contains("\\");
+            return new[] { '*', '.', ' ', '\\', '/', ':', '?', '<', '>', '|', '"' };
```

Comment:
I have included the blank space as invalid character for a package in order to keep the previous implementation of the function, notice there are already created packages that contain blank spaces.

Reply:
Is there a .NET API which can return these?

Reply:
So, I think this could be problematic, what happens when a user wants to publish a new version for a package that already has spaces in the name, will that get flagged?
Also, why are we invalidating spaces?

Reply:
> Is there a .NET API which can return these?

I was tented to use the function Path.GetInvalidFileNameChars but it returns more characters that we need

![imagen](https://user-images.githubusercontent.com/96534278/163464102-3e39ca9e-1945-42e8-a8a4-dd9236abbc76.png)

Reply:
> So, I think this could be problematic, what happens when a user wants to publish a new version for a package that already has spaces in the name, will that get flagged? Also, why are we invalidating spaces?

As I mention, the previous version of the function invalids the space, I will remove it.

Reply:
The `Path.GetInvalidFileNameChars` looks solid to me, and it seems it includes space as well? @zeusongit I am confused, Do we allow space before even though it was included as the invalid char check?

Reply:
@QilongTang Actually the ContainsSpecialCharacters function hadn't been used before this branch, at least in Dynamo solution.

Reply:
Perfect, let's replace it

Reply:
Done